share/build-navigation/build-navigation-json.xsl

```diff
@@ -80,39 +86,218 @@
+        </xsl:call-template>
+      </xsl:otherwise>
+    </xsl:choose>
+  </xsl:template>
+
+
+  <xsl:template name="lifecycle-json">
+    <xsl:param name="nodes"/>
+    <xsl:param name="lifecycle"/>
+    <xsl:param name="site"/>
+    <xsl:variable name="filename" select="concat($output_root, $site, '-', $lifecycle, '.json')"/>
+    <xsl:variable name="list">
+      <xsl:choose>
+        <xsl:when test="$lifecycle='supported'">
+          <xsl:value-of select="$existing-sets-supported"/>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:value-of select="$existing-sets-unsupported"/>
+        </xsl:otherwise>
+      </xsl:choose>
+    </xsl:variable>
+
+   <xsl:message>lifecycle-json:
+     JSON file="<xsl:value-of select="$filename"/>"
+     nodes=<xsl:value-of select="count($nodes)"/>
+     all product nodes=<xsl:value-of select="count(//product)"/>
+     current node=<xsl:value-of select="local-name(.)"/>
+     list=<xsl:value-of select="$list"/>
+   </xsl:message>
+
     <exsl:document
-      href="{$output_root}unsupported.json"
+      href="{$filename}"
       method="text"
       encoding="UTF-8"
       indent="no"
       media-type="application/x-json">
 {
+  "site-section": "<xsl:value-of select="$site"/>",
   "productline": {
-     <xsl:apply-templates select="//product" mode="generate-productline-list">
+     <xsl:apply-templates select="$nodes" mode="generate-productline-list">
        <xsl:sort
          lang="en"
          select="normalize-space(translate((name|sortname)[last()], '&sortlower;', '&sortupper;'))"/>
-       <xsl:with-param name="list" select="$existing-sets-unsupported"/>
+       <xsl:with-param name="list" select="$list"/>
      </xsl:apply-templates>
   },
   "product": {
-     <xsl:apply-templates select="//product" mode="generate-product-list">
+     <xsl:apply-templates select="$nodes" mode="generate-product-list">
        <xsl:sort
          lang="en"
          select="normalize-space(translate((name|sortname)[last()],'&sortlower;', '&sortupper;'))"/>
-       <xsl:with-param name="list" select="$existing-sets-unsupported"/>
+       <xsl:with-param name="list" select="$list"/>
      </xsl:apply-templates>
   }
 }
+</exsl:document>
+  </xsl:template>
+
+
+  <xsl:template match="*" mode="create-site-section-json">
+    <xsl:variable name="filename" select="concat($output_root, 'site-section.json')"/>
+
+    <xsl:message>*** create-site-section-json: "<xsl:value-of select="$site_sections"/>"</xsl:message>
+    <exsl:document
+      href="{$filename}"
+      method="text"
+      encoding="UTF-8"
+      indent="no"
+      media-type="application/x-json">
+      <xsl:text>[&#10;</xsl:text>
+
+      <xsl:call-template name="find-all-json-files">
+        <xsl:with-param name="sites" select="concat($site_sections, ' ')"/>
+        <!-- Remove double entries -->
+        <xsl:with-param name="lifecycle"
+          select="//*[@lifecycle][not(@lifecycle=preceding::*/@lifecycle)]"/>
```

Comment:
I am a bit unsure why you need to "find JSON files" here and why there is a need for the entire template below -- you just need to recursively output the space-separated content of $site_sections, afaict. The JSON output I'd expect is something like this for the first iteration: `{'main': {'default': true}, 'sbp': {'default': false} }`.

Later on, this may be expanded to something that includes tab titles and their translations etc., but right now, we have no config to add the translation to, but I hope that changes after #259 is fixed.

Reply:
If I got this right, the output should be covered in 8cfd97b. Some cleanup of uncommented code is necessary.